hyperspy/drawing/marker.py

```diff
@@ -0,0 +1,288 @@
+
+    @property
+    def marker_properties(self):
+        return self._marker_properties
+
+    @marker_properties.setter
+    def marker_properties(self, kwargs):
+        if 'type' in kwargs:
+            self.type = kwargs['type']
+            del kwargs['type']
+
+        for key, item in kwargs.iteritems():
+            if item is None and key in self._marker_properties:
+                del self._marker_properties[key]
+            else:
+                self._marker_properties[key] = item
+        if self.marker is not None:
+            plt.setp(self.marker, **self.marker_properties)
+            try:
+                # self.ax.figure.canvas.draw()
+                self.ax.hspy_fig._draw_animated()
+            except:
+                pass
+
+    def set_marker_properties(self, **kwargs):
+        self.marker_properties = kwargs
+
+    def plot(self):
+        if self.type == 'text':
+            self.marker = self.ax.text(
+                self.get_data_position('x1'), self.get_data_position('y1'),
+                self.get_data_position('text'), **self.marker_properties)
+        elif self.type == 'line':
+            self.marker = self.ax.vlines(0, 0, 1,
+                                         **self.marker_properties)
+            self.set_line_segment()
+        elif self.type == 'axvline':
+            self.marker = self.ax.axvline(self.get_data_position('x1'),
+                                          **self.marker_properties)
+        elif self.type == 'axhline':
+            self.marker = self.ax.axhline(self.get_data_position('y1'),
+                                          **self.marker_properties)
+        elif self.type == 'pointer':
+            self.marker = self.ax.scatter(self.get_data_position('x1'),
+                                          self.get_data_position(
+                                              'y1'),
+                                          **self.marker_properties)
+            if self.get_data_position('size') is None:
+                self.set_data(size=20)
+            self.marker._sizes = [self.get_data_position('size')]
+
+        elif self.type == 'rect':
+            width = abs(self.get_data_position('x1') -
+                        self.get_data_position('x2'))
+            height = abs(self.get_data_position('y1') -
+                         self.get_data_position('y2'))
+            self.marker = self.ax.add_patch(plt.Rectangle(
+                (self.get_data_position('x1'), self.get_data_position('y1')),
+                width, height, **self.marker_properties))
+
+        self.marker.set_animated(True)
+        # To be discussed, done in Spectrum figure once.
+        # self.axes_manager.connect(self.update)
+        try:
+            self.ax.hspy_fig._draw_animated()
+            # self.ax.figure.canvas.draw()
+        except:
+            pass
+
+    def set_data(self, x1=None, y1=None,
+                 x2=None, y2=None, text=None, size=None):
```

Comment:
The function is not fail-safe - i.e. when only passing a single argument, it overwrites the rest. I suggest adding "overwrite" or "update" flag, which only overwrites the values of fields that are not ```None```.

That's the reason why plotting 'pointer' fails now - if no size given, you set the default to 20, but at the same time overwrite all the other fields (i.e. x1 and y1)